tf2_eigen/include/tf2_eigen/tf2_eigen.h

```diff
@@ -186,10 +314,58 @@ geometry_msgs::msg::Pose toMsg(const Eigen::Affine3d& in) {
   msg.position.x = in.translation().x();
   msg.position.y = in.translation().y();
   msg.position.z = in.translation().z();
-  msg.orientation.x = Eigen::Quaterniond(in.rotation()).x();
-  msg.orientation.y = Eigen::Quaterniond(in.rotation()).y();
-  msg.orientation.z = Eigen::Quaterniond(in.rotation()).z();
-  msg.orientation.w = Eigen::Quaterniond(in.rotation()).w();
+  Eigen::Quaterniond q(in.linear());
+  msg.orientation.x = q.x();
+  msg.orientation.y = q.y();
+  msg.orientation.z = q.z();
+  msg.orientation.w = q.w();
+  if (msg.orientation.w < 0) {
```

Comment:
Renormalizing the quaternion here will break the the one to one mapping I'd expect from a conversions. A -> B -> A should return the same thing.